cartridges/int_algolia/cartridge/scripts/algolia/recommend/utils.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+var MASTER_LEVEL = 'master-level';
+
+/**
+ * @description Set anchor products for the current session based on the current record model
+ * @returns {void}
+ */
+function anchorProductSetter() {
+    var productIds = session.privacy.algoliaAnchorProducts;
+    if (!productIds || productIds.length === 0) {
+        return;
+    }
+
+    var algoliaData = require('*/cartridge/scripts/algolia/lib/algoliaData');
+    var recordModel = algoliaData.getPreference('RecordModel');
+    if (recordModel !== MASTER_LEVEL) {
+        return;
+    }
+
+    var productMgr = require('dw/catalog/ProductMgr');
+    var algoliaAnchorProducts = [];
+
+    productIds = JSON.parse(productIds);
+    for (var i = 0; i < productIds.length; i++) {
+        var productId = productIds[i];
+        var product = productMgr.getProduct(productId);
+        if (product) {
+            var masterProduct = product.isVariant() ? product.getMasterProduct() : product;
+            algoliaAnchorProducts.push(masterProduct.ID);
+        }
+    }
+
+    session.privacy.algoliaAnchorProducts = JSON.stringify(algoliaAnchorProducts);
```

Comment:
This will work only in a ISML context. Can you make this function a pure function? Which take an array of productIds and return an array of master IDs, without modifying external things, like:
```
function getMasterProductIds(productIDs) {
  var masterProductIDs = [];
  // ...
  return masterProductIDs;
}
```

Reply:
I also think that I can create a separate function for here. because it is code repetition on all templates. What do you think?

![Image2024-03-21 at 10 21 59](https://github.com/algolia/algoliasearch-sfcc-b2c/assets/36648230/d3fdf93e-efe1-4063-9c5a-e96f0fa2d823)

Reply:
Oh sure the test about the recordModel can probably be put in the helper function too :+1: